### PR TITLE
Fixes crash with prompt-bottom, code cleanup and move help/status label

### LINKED ISSUE #185: An entry is highlighted even if patter selection is empty
Reproducible as follows:
* write a patter that no history entry matches i.e. selection is empty
* use <kbd>UP</kbd>/<kbd>DOWN</kbd> keys (repeatedly)
* although there is empty selection, `hh` unexpectedly shows a highlighted entry

### PR DESCRIPTION
**Pull request by pknowles**
Take all this with a grain of salt. If any is useful, cherry pick away. The most useful patch I think is the selectionCursorPosition one, which I'm hoping fixes #185. However I'm sure there are features I haven't tested.

My preference for moving status messages into the one line, optionally removing the help line and shifting both to the top for prompt-bottom are pretty drastic and not covered by config options. Just a suggestion.

### CODE REVIEW

File: src/hstr.c [hunk 1]
```diff
@@ -1312,16 +1269,21 @@ void loop_to_select(Hstr *hstr)
...
                     print_pattern(pattern, hstr->promptY, basex);
                     cursorX=getcurx(stdscr);
                     cursorY=getcury(stdscr);
                 }
 
-                result = hstr_print_selection(maxHistoryItems, pattern, hstr);
+                previousSelectionCursorPosition = SELECTION_CURSOR_IN_PROMPT;
+                result = hstr_make_and_print_selection(pattern, hstr);
+                if (hstr->selectionSize > 0) {
+                    selectionCursorPosition=0;
```
Comment (pknowles) [inline src/hstr.c]: I should revert this. I was thinking it might be nice to have the first matching entry always selected to be more explicit about what you get when you hit enter, but should be separated into another patch.

**Comment by dvorka** [comment]
This is great contribution! I merge cherry picked code later today :wink: Thanks!

File: src/hstr.c [hunk 2]
```diff
@@ -328,73 +329,53 @@ unsigned recalculate_max_history_items()
...
+    const char *hstr_config=getenv(HH_ENV_VAR_CONFIG);
+    if (!hstr_config || !strlen(hstr_config)) {
+        return;
+    }
+    size_t const len = strlen(hstr_config)+1;
+    char* config_items = malloc(len);
+    memcpy(config_items, hstr_config, len);
+    char* saveptr = config_items;
+    char* item;
+    while ((item = strtok_r(saveptr, ",", &saveptr)) != NULL) {
```
Comment (dvorka) [inline src/hstr.c]: Unfortunately refactoring of if statements in this loop is wrong - there is a semantics in which option can overwrite other and making this flat kills the logic that was implemented in there. Thus I cannot merge this particular change.

File: src/hstr.c [hunk 3]
```diff
@@ -328,73 +329,53 @@ unsigned recalculate_max_history_items()
...
-        if(strstr(hstr_config,HH_CONFIG_MINIMAL)) {
+        } else if(strcmp(item,HH_CONFIG_MINIMAL)==0) {
             hstr->showHelp=false;
+        } else {
+            fprintf(stderr, "WARNING: unknown item in %s: %s\n", HH_ENV_VAR_CONFIG, item);
         }
-
-        recalculate_max_history_items();
     }
+    free(config_items);
```
Comment (dvorka) [inline src/hstr.c]: @dvorka Memory leak to be fixed.

**Comment by dvorka** [comment]
I need to merge this contribution to a branch and test/process polish it there.

**Comment by dvorka** [comment]
@pknowles I have created #221 to process your contribution in branch - I will continue there. Just in case you would have additional code/additions use the branch for pull requests. Thanks again for you work!
